processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -504,26 +508,28 @@ private long getMaxTimeMillis()
     return getFacts().lastKey().getTimestamp();
   }
 
-  private String[] getDimVals(final DimDim dimLookup, final List<String> dimValues)
+  private int[] getDimVals(final DimDim dimLookup, final List<String> dimValues)
   {
-    final String[] retVal = new String[dimValues.size()];
     if (dimValues.size() == 0) {
       // NULL VALUE
-      if (!dimLookup.contains(null)) {
-        dimLookup.add(null);
-      }
+      dimLookup.add("");
       return null;
     }
+
+    if (dimValues.size() == 1) {
+      return new int[]{dimLookup.add(dimValues.get(0))};
+    }
+
+    // todo: make option for this
```

Comment:
Thinking of doing it later. We need some dimension spec (not capability) which can be used for describing it in more detail (something like types, options, comparators, null-handling, etc.) which is currently not available.